### PR TITLE
feat: add devWorkflow provider

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
add rbWorkflow provider

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTkwXSwiYXNzaWduZWUiOlsiMTAwMTI2MSJdfQ%3D%3D&id=308026&iterationID=1190&pId=305580&type=TASK)

#### Specified Reviewers:

/assign @Effet 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     add rbWorkflow provider         |
| 🇨🇳 中文    |      增加研发工作流        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/dop/providers/devworkflow/devworkflow.service.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+func (s *ServiceImplement) InitWorkFlows() db.WorkFlows {
+	return db.WorkFlows{
+		{
+			Name:             "DEV",
+			FlowType:         "two_branch",
+			TargetBranch:     "develop",
+			ChangeFromBranch: "",
+			ChangeBranch:     "feature/*,bugfix/*",
+			EnableAutoMerge:  true,
+			AutoMergeBranch:  "dev/temp/*",
```
Comment (Effet) [inline modules/dop/providers/devworkflow/devworkflow.service.go:87]: AutoMergeBranch for `DEV` flow should be `dev` as default, it just one branch.

File: modules/dop/providers/devworkflow/devworkflow.service.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+			StartWorkflowHints: nil,
+		},
+		{
+			Name:               "STAGING",
+			FlowType:           "three_branch",
+			TargetBranch:       "master",
+			ChangeFromBranch:   "develop",
+			ChangeBranch:       "release/*",
+			EnableAutoMerge:    true,
+			AutoMergeBranch:    "staging/temp/*",
```
Comment (Effet) [inline modules/dop/providers/devworkflow/devworkflow.service.go:120]: `STAGING` flow should `EnableAutoMerge: false` and have no AutoMergeBranch

File: pkg/erda-configs/permission/backend/sys/org/project/project.yml [hunk 3]
```diff
@@ -839,4 +839,15 @@
   resource: testReport
   action: UPDATE
 
-## test report end ##
+  ## test report end ##
```
Comment (Effet) [inline pkg/erda-configs/permission/backend/sys/org/project/project.yml:842]: bad indent

File: pkg/erda-configs/permission/backend/sys/org/project/project.yml [hunk 4]
```diff
@@ -839,4 +839,15 @@
...
+
+## rb workflow start ##
```
Comment (Effet) [inline pkg/erda-configs/permission/backend/sys/org/project/project.yml:844]: rb -> dev

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/dop/providers/devworkflow/devworkflow.service.go (hunk 1 above)
Comment (littlejiancc) [reply modules/dop/providers/devworkflow/devworkflow.service.go:87]: done

Comment (littlejiancc) [reply modules/dop/providers/devworkflow/devworkflow.service.go:87]: done

File: modules/dop/providers/devworkflow/devworkflow.service.go (hunk 2 above)
Comment (littlejiancc) [reply modules/dop/providers/devworkflow/devworkflow.service.go:120]: done

File: pkg/erda-configs/permission/backend/sys/org/project/project.yml (hunk 4 above)
Comment (littlejiancc) [reply pkg/erda-configs/permission/backend/sys/org/project/project.yml:844]: done

**Review by Effet — DISMISSED** [review]

**Review by Effet — APPROVED** [review]

**Pull request merged**